### ISSUE TITLE
Image Scaling and Anti-Aliasing Issues on High-DPI Displays with Display Scaling Enabled

### ISSUE THREAD

**Issue by mahdishalchian** [nomacs/nomacs#1225]
Issue Description:
When using nomacs on a monitor with a resolution larger than 1920x1080 (e.g., 4K monitors), adjusting the display scaling settings (e.g., setting scaling to 150%) causes the following issues with image rendering:

    Images displayed at 100% (1:1) scale do not have the correct dimensions and appear larger than their actual size.
    Images lose their anti-aliasing and become pixelated.

These issues seem to be related to how nomacs handles scaling on high-DPI displays.

Steps to Reproduce:

    Use a high-resolution monitor (e.g., 4K).
    Increase the display scaling in your operating system (e.g., set scaling to 150%).
    Open any image in nomacs.
    Set the image scale to 100% (1:1).

Expected Behavior:

    The image should appear at its correct dimensions.
    The image should retain smooth anti-aliasing without pixelation.

Observed Behavior:

    The image appears larger than its actual dimensions.
    The image becomes pixelated and loses anti-aliasing.

Environment Details:

    Operating System: [Windows 11]
    Nomacs Version: [3.19.1]
    Monitor Resolution: [3840x2160]
    Display Scaling: [150%]
    
My Display scale:
![image](https://github.com/user-attachments/assets/7a3f17a5-0f47-473e-984c-e0b6eb9f7de4)

Comparison between nomacs and windows photos, image size 100%:
![image](https://github.com/user-attachments/assets/00257185-0bf1-4fd8-b924-a80bf016ab54)

Comparison between nomacs and windows photos, anti-aliasing:
nomacs:
![image](https://github.com/user-attachments/assets/5b4cea8c-8b50-4012-9663-648cbe5d5f5d)
windows:
![image](https://github.com/user-attachments/assets/1aaac1c8-2b53-472c-9685-4795cd017003)

**Comment by scrubbbbs** [comment]
Thanks for reporting, I was not aware that dpi scaling was also broken on windows. Related: #1126 

#385 notes that scaling couldl be incorrect on  multi-monitor setups since nomacs takes the dpi from the highest-dpi screen, and not the screen the window is currently occupying. Do you have multiple monitors by chance?

**Comment by mahdishalchian** [comment]
Yes, I am using two monitors, but both have the same specifications:
	•	Resolution: 4K
	•	Display Scaling: 150%

**Comment by scrubbbbs** [comment]
> Yes, I am using two monitors, but both have the same specifications:
> • Resolution: 4K
> • Display Scaling: 150%

The width/height (physical) plus the resolution determines the DPI. Are they the same physical dimension too?

**Comment by mahdishalchian** [comment]
Both monitors have the same physical dimensions and are exactly the same brand and model.

**Comment by scrubbbbs** [comment]
I can confirm this on windows10 and 11. It seems display scaling is not taken into account on any platform, which makes this a duplicate of #1126. 

The same workaround in #1126 applies on windows (to recent builds anyways, not sure about 3.17 etc). We can scale the UI with the font dpi (144=96*1.5) and compensate for the display scaling by shrinking (0.666=1/1.5)

For example, in a batch  file or cmd.exe from the nomacs program directory:

```batch
set QT_FONT_DPI=144
set QT_SCALE_FACTOR=0.66666
nomacs.exe
```

**Comment by mahdishalchian** [comment]
I don’t fully understand how to fix this issue. Where exactly should I enter these values?

**Comment by scrubbbbs** [comment]
These are the batch file/console commands needed  to undo scaling in nomacs. It isn't a proper fix; for example, even if you did start nomacs as shown, if you open a file from explorer it will behave as before. 

Someone more knowledgeable on batch files might be able to craft an actual solution.

**Comment by scrubbbbs** [comment]
Fixed with #1382
